libraries/render-utils/src/RenderShadowTask.cpp

```diff
@@ -318,16 +317,76 @@ RenderShadowSetup::RenderShadowSetup() :
 }
 
 void RenderShadowSetup::configure(const Config& configuration) {
-    setConstantBias(0, configuration.constantBias0);
-    setSlopeBias(0, configuration.slopeBias0);
-#if SHADOW_CASCADE_MAX_COUNT>1
-    setConstantBias(1, configuration.constantBias1);
-    setSlopeBias(1, configuration.slopeBias1);
-    setConstantBias(2, configuration.constantBias2);
-    setSlopeBias(2, configuration.slopeBias2);
-    setConstantBias(3, configuration.constantBias3);
-    setSlopeBias(3, configuration.slopeBias3);
-#endif
+    distanceTriggeredByConfig = _globalMaxDistance != configuration.globalMaxDistance;
+    biasTriggeredByConfig = _biasInput != configuration.biasInput;
+
+    // go back to using the config's default bias values if a change to any of those is triggered
+    if (constant0 != configuration.constantBias0 || slope0 != configuration.slopeBias0 ||
+        constant1 != configuration.constantBias1 || slope1 != configuration.slopeBias1 ||
+        constant2 != configuration.constantBias2 || slope2 != configuration.slopeBias2 || 
+        constant3 != configuration.constantBias3 || slope3 != configuration.slopeBias3) {
+        constant0 = configuration.constantBias0;
+        slope0 = configuration.slopeBias0;
+        constant1 = configuration.constantBias1;
+        slope1 = configuration.slopeBias1;
+        constant2 = configuration.constantBias2;
+        slope2 = configuration.slopeBias2;
+        constant3 = configuration.constantBias3;
+        slope3 = configuration.slopeBias3;
+        changeInDefaultConfigValues = true;
+        distanceTriggeredByConfig = false;
+        biasTriggeredByConfig = false;
+
+        setConstantBias(0, constant0);
+        setSlopeBias(0, slope0);
+
+        #if SHADOW_CASCADE_MAX_COUNT > 1
+            setConstantBias(1, constant1);
+            setConstantBias(2, constant2);
+            setConstantBias(3, constant3);
+
+            setSlopeBias(1, slope1);
+            setSlopeBias(2, slope2);
+            setSlopeBias(3, slope3);
+        #endif
+    }
+        
+    // modify bias using single input and work in calculateBias()
+    if (distanceTriggeredByConfig) {
+        changeInDefaultConfigValues = false;
+        _globalMaxDistance = configuration.globalMaxDistance;
+        calculateBiases();
+    }
+    if (biasTriggeredByConfig) {
+        changeInDefaultConfigValues = false;
+        _biasInput = configuration.biasInput;
+        calculateBiases();
+    }
+}
+
+void RenderShadowSetup::calculateBiases() {
+    // slope scaling values derived from ratio between original constantBias and slopeBias pairs
+    const std::vector<float> SLOPE_SCALES = {2.7f, 3.0f, 3.7f, 3.5f};
```

Comment:
```suggestion
    const std::array<float, 4> SLOPE_SCALES = { 2.7f, 3.0f, 3.7f, 3.5f };
```
(the other std::vectors should also be arrays)